docs/src/components/MobileNav.tsx

```diff
@@ -119,6 +119,28 @@ export function Popup({ children, className, ...props }: Dialog.Popup.Props) {
   );
 }
 
+// iOS Safari appears to flicker the scroll containers during tasking animations
+// As a workaround, we set an attribute on body to know when the nav is mounted
+function MountEffect() {
+  const timeout = React.useRef(0);
+
+  React.useLayoutEffect(() => {
+    window.clearTimeout(timeout.current);
+    document.body.setAttribute('data-mobile-nav-open', '');
+
+    return () => {
+      window.clearTimeout(timeout.current);
+      timeout.current = window.setTimeout(() => {
+        document.body.removeAttribute('data-mobile-nav-open');
+        // Safari seems to need some arbitrary time to be done with whatever causes the flicker
+        // Using setTimeout 0, RAFs, or even double RAFs doesn't work reliably
+      }, 100);
```

Comment:
Setting an arbitrary timeout makes me incredibly angry, but I can't find anything deterministic that works instead.

The effect runs right before the Popup is unmounted, so the animations are done by that time.

Any ideas?

Reply:
Crazy suggestion, have you tried `setTimeout 0.5`?

Reply:
@mj12albert small delays like `0.5` work about half the time for me 🤷‍♂️